test/Feature/UpsertSpec.hs

```diff
@@ -233,87 +241,132 @@ spec =
 
       context "Inserting row" $ do
         it "succeeds on table with single pk col" $ do
-          get "/tiobe_pls?name=eq.Go" `shouldRespondWith` "[]"
-          put "/tiobe_pls?name=eq.Go" [json| [ { "name": "Go", "rank": 19 } ]|] `shouldRespondWith` 204
-          get "/tiobe_pls?name=eq.Go" `shouldRespondWith` [json| [ { "name": "Go", "rank": 19 } ]|] { matchHeaders = [matchContentTypeJson] }
+          -- assert that the next request will indeed be an insert
```

Comment:
Most of the `PUT` tests had some read requests before the main request to make sure that, what was going to happen next, would indeed be either an `INSERT` or `UPDATE` effectively. I added comments to those as an explanation and also added some of those requests, where they were missing.